src/Streamly/Internal/Data/Parser/ParserD/Types.hs

```diff
@@ -505,6 +499,7 @@ splitSome (Fold fstep finitial fextract) (Parser step1 initial1 extract1) =
     {-# INLINE step #-}
     step (Tuple3' st cnt (Left fs)) a = do
         r <- step1 st a
+        -- In the Left state, count is used only for the assert
```

Comment:
You can remove this. This remained when I removed `Partial1` and `Done1`.

Reply:
Let's keep the assert for now, its good for debugging.